src/main/java/com/rogerr/custom/controller/UserController.java

```diff
@@ -0,0 +1,120 @@
+package com.rogerr.custom.controller;
+ 
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import com.rogerr.custom.model.User;
+import com.rogerr.custom.service.UserService;
+
+
+@RestController
+public class UserController {
+
+    @Autowired
+    private UserService userService;
+
+    private static Logger logger = LoggerFactory.getLogger(UserController.class);
+
+    // -------------------Retrieve All Users--
+
+    @RequestMapping(value = "/user/", method = RequestMethod.GET)
+    public ResponseEntity<?> listAllUsers() {
+        List<User> users = userService.findAllUsers();
+        if (users.isEmpty()) {
+            return new ResponseEntity<List<User>>(HttpStatus.NO_CONTENT);
+        }
+        return new ResponseEntity<List<User>>(users, HttpStatus.OK);
+    }
+
+    // -------------------Retrieve Single User--
+
+    @RequestMapping(value = "/user/{id}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> getUser(@PathVariable("id") long id) {
+        logger.debug("Fetching User with id " + id);
+        User user = userService.findById(id);
+        if (user == null) {
+            logger.debug("User with id " + id + " not found");
+            return new ResponseEntity<User>(HttpStatus.NO_CONTENT);
+        }
+        return new ResponseEntity<User>(user, HttpStatus.OK);
+    }
+
+    // -------------------Create a User--
+
+    @RequestMapping(value = "/user/", method = RequestMethod.POST)
+    public ResponseEntity<?> createUser(@RequestBody User user, UriComponentsBuilder ucBuilder) {
+        logger.debug("Creating User " + user.getUsername());
+
+        if (userService.isUserExist(user)) {
+            logger.debug("A User with name " + user.getUsername() + " already exist");
+            return new ResponseEntity<Void>(HttpStatus.CONFLICT);
+        }
+
+        userService.saveUser(user);
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setLocation(ucBuilder.path("/user/{id}").buildAndExpand(user.getId()).toUri());
+        return new ResponseEntity<Void>(headers, HttpStatus.CREATED);
```

Comment:
install an IDE-based linter like Sonarlint or FindBugs.  I don't think specifying void type is necessary, just <>